### DISCUSSION TITLE
aiohttp Parallel Web Server

### DISCUSSION THREAD

**Discussion by redradist** [aio-libs/aiohttp#7305] [Q&A]
I am interesting in making a parallel aiohttp web.Application server
Fir this I have tried to use the following code:
```python
reuseport = True
runner = web.AppRunner(app)
await runner.setup()
sock = mk_socket(host=args.domain, port=args.port, reuseport=reuseport)
site = web.SockSite(runner=runner, sock=sock, ssl_context=ssl_context)
await site.start()
print(f'Server started at https://{args.domain}:{args.port}')
return site, app, runner
```
I took this code from https://stackoverflow.com/a/59219359/5157245

But looks like this solution hang out after working some time ...
After debugging it looks like it is hanging out on `select` function of module `selectors`

Is there a working example of parallel aiohttp web.Application server ?
Maybe in this example I am doing something wrong

**Comment by Dreamsorcerer** [comment]
> After debugging it looks like it is hanging out on `select` function of module `selectors`

That's not hanging, that's waiting for activity on the port.

> Is there a working example of parallel aiohttp web.Application server ? Maybe in this example I am doing something wrong

No, this is a rather unusual thing to do, so I doubt there are any official examples around.
A lot more sensible approach is to just run multiple application instances behind nginx or another load balancer.
